docs/spec.md

```diff
@@ -2633,6 +2633,52 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
+
+Fills an `output` tensor with values in increasing order starting from zero
+along the `iota_dimension` dimension. `output` tensor shape is specified
+dynamically via `output_shape` input. More formally,
+
+`output[result_index] = constant(is_quantized(output) ?
+quantize(result_index[iota_dimension], element_type(output)) :
+result_index[iota_dimension], element_type(output))`.
+
+#### Inputs
+
+| Label | Name             | Type                                                                               | Constraints |
+|-------|------------------|------------------------------------------------------------------------------------|-------------|
+| (I1)  | `iota_dimension` | `si64`                                                                             | (C1)        |
+| (I2)  | `output_shape`   | 1-dimensional tensor constant of type `si64`                                       | (C1), (C2)  |
```

Comment:
naming: `output_shape` or just  `shape` ?

Reply:
Should be [`output_shape`](https://github.com/openxla/stablehlo/blob/205f405613a14554095d678ebd87c1ced924f7af/stablehlo/dialect/StablehloOps.td#L140).

Reply:
thanks for confirming. Yeh, I used `output_shape` from the op def.

In general, we can always go back and update the op def as per the spec?

Reply:
from other discussion - 
"Changing name has implications for serialization and we have a ticket to track this https://github.com/openxla/stablehlo/issues/351."